[Feat] 사용자 쿠키가 반영된 API 연결 & 화면 업데이트

Jinsujin opened the pull request:
# 💡 Issue
- CoreData 를 앱 초기화할때 연결한다

# 📝 Tasks
- [x] CoreData 연결
- [x] 사용자쿠키값을 앱 초기화할때 받아온다
- [x] 사용자 쿠키값을 반영한 전체 데이터 get API 호출
- [x] 응답을 CoreData 에 저장 (sync)
- [x] 가져온 전체 데이터를 화면에 반영

## CoreData 연결

### 🔹 고민 => 해결!
- coreData 에서 데이터를 읽어와서 화면에 뿌려주는 곳은 CardDataMiddleWare 일까요?
- 현재 네트워크에서 불러오는 작업이 CardDataMiddleWare.fetchAllCards 메서드에 정의가 되어 있는데
- 네트워크에서 불러오고 나서 -> 불러온 값을 core data 에 저장 -> fetchResultInBoard 멤버변수에 들고온 데이터를 set -> Noti를 통해 화면이 업데이트 된다
- 이 순서가 맞을까요?

## API 로 가져온 데이터를 CoreData 에 반영
### 🔹 고민
- API 에서 가져온 데이터를 앱 내부(CoreData) 와 비교해 싱크를 맞춰줘야 하는 작업
- 현재 id 값만 비교해 중복된 id 가 CoreData 에 있으면 CoreData 에 저장하지 않는 로직입니다
- 만약에 필드값이 하나라도 다르면 서버에서 가져온 데이터를 CoreData 에 반영해야 합니다
- 필드값 하나하나를 비교해 다른 데이터인지 판별하는 방법은 비효율적이라는 생각이 듭니다
- 싱크를 어떻게 맞춰야 할지 고민입니다

Jinsujin committed:
feat: CoreData 연결 (#23)

SceneDelegate 에서 Root View Controller 에 메서드를 통해 주입
CoreData 연결 확인함

SangHwi-Back commented:
안녕하세요 로사 새벽에 메일이 와 있었네요! 위의 고민으로 적어주신 부분은 제 생각도 동일합니다.

1. CardDataMiddleWare에서 REST-API 요청
2. CoreData 초기화 후 받아온 데이터 저장(네트워크 요청 실패할 경우 CoreData 이용)
3. NotificationCenter.default.post()를 이용해서 각 뷰 컨트롤러 업데이트

입니다!

Jinsujin committed:
feat:  PersistenceProvider 싱글톤 제거, CardDataMiddleWare 에 주입  (#23)

Jinsujin committed:
feat: 유저 쿠키를 사용해 전체 데이터를 불러오는 API 호출, 뷰에 반영 (#23)

1. 실제 사용자 쿠키를 앱 실행시(AppDelegate.didFinishLaunching) 얻어 온다
2. Debug 용 api 호출이 아닌, 사용자 쿠키를 담은 API 를 호출한다
3. 응답으로 받은 전체 카드 데이터를 MiddleWare 에 반영
4. 화면 업데이트
화면에서 뿌려줄 모델을 CardData -> Card 로 변경
Card 모델은 앱 내부에서만 사용될 모델의 역할
CardData 모델은 API를 이용할때만 사용

SangHwi-Back commented on Client/iOS/TodoList/View/TodoRepositoryRespondWrapper.swift:
```diff
@@ -9,7 +9,7 @@ class TodoRepositoryRespondWrapper: NSObject {
     var todoBoard: TodoBoard
     
-    private(set) var dataSource = [CardData]()
+    private(set) var dataSource = [Card]()
```
테이블 뷰의 데이터 타입이 결국은 바뀌게 되었군요 ㅠㅠ

CardData 를 사용하여 만든 클래스라 잘 작동할지는 모르겠습니다.
카드 데이터를 추가하는 부분도 있으니 머지 진행하고 뷰가 이상 없는지 확인해보도록 하겠습니다.

SangHwi-Back commented:
로사 더 수정하실 부분이 없다면 이제 머지를 하고 제 코드도 머지 진행하려고 합니다. 괜찮을까요??
